File: ServeSys/ServeSys.API/Modules/Identity/DTOs/AuthResponse.cs
namespace ServeSys.API.Modules.Identity.DTOs;

public class AuthResponse
{
    public string Token { get; set; } = string.Empty;
    public string RefreshToken { get; set; } = string.Empty;
    public DateTime Expiration { get; set; }
    public string? Email { get; set; }
    public string FullName { get; set; } = string.Empty;
}

File: ServeSys.Mobile/servesys/lib/features/auth/data/models/auth_response.dart
class AuthResponseData {
  final String token;
  final String refreshToken;
  final DateTime expiration;
  final String? email;
  final String fullName;

  AuthResponseData({
    required this.token,
    required this.refreshToken,
    required this.expiration,
    this.email,
    required this.fullName,
  });

  factory AuthResponseData.fromJson(Map<String, dynamic> json) {
    return AuthResponseData(
      token: json['token'] as String,
      refreshToken: json['refreshToken'] as String,
      expiration: DateTime.parse(json['expiration'] as String).toUtc(),
      email: json['email'] as String?,
      fullName: json['fullName'] as String,
    );
  }
}
